akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -44,6 +44,16 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
 
       def render(h: HttpHeader) = r ~~ h ~~ CrLf
 
+      val hostString: InetSocketAddress ⇒ String = {
+        // Retrieve the original host string that was given (IP or DNS name) if running Java 7 or later
+        // using the getHostString() method. This avoids a reverse DNS query from calling getHostName()
+        // if the original host string is an IP address.
+        val clazz = classOf[InetSocketAddress]
+        address ⇒ clazz.getMethods.collectFirst {
+          case m if m.getName == "getHostString" => m.invoke(address)
+        }.getOrElse(address.getHostName).asInstanceOf[String]
+      }
```

Comment:
With a slight modification it is possible to avoid doingthe method lookup for each call:

~~~scala
val hostString: InetSocketAddress => String =
  try {
    val m = classOf[InetSocketAddress].getDeclaredMethod("getHostString")
    require(m.getReturnType == classOf[String])
    require(m.getParameterCount == 0)
    address => m.invoke(address).asInstanceOf[String]
  } catch {
    case NonFatal(_) =>
    address => address.getHostName()
  }
~~~